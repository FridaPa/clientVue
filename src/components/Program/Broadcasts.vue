<template>
  <v-row justify="center">
    <v-dialog v-model="dialog" persistent max-width="700px">
      <template v-slot:activator="{on}">
        <button class="btnDialogAccept" @click="dialog = true">Se sändningar</button>
      </template>
      <v-card>
        <v-card-title>
          <span class="getBroadcasts">Sändningar för {{title}} säsong {{season}} episod {{episode}}</span>
        </v-card-title>
        <v-card-text>
          <v-container>
            <v-col cols="10">
              <button
                class="preliminary"
                v-bind:class="{ activeButton: (this.active == 'preliminary') }"
                v-on:click="makeActive('preliminary')"
              >Preliminära sändningar</button>
              <button
                class="actual"
                v-bind:class="{ activeButton: (this.active == 'actual') }"
                v-on:click="makeActive('actual')"
              >Faktiska sändningar</button>
              <v-toolbar class="toolbar" color="#177CC4" height="50px">
                <v-row>
                  <v-col cols="3">
                    <p class="toolbarFont">Kanal</p>
                  </v-col>
                  <v-col cols="3">
                    <p class="toolbarFont">Sändningsdag</p>
                  </v-col>
                  <v-col cols="3">
                    <p class="toolbarFont">Starttid</p>
                  </v-col>
                  <v-col cols="3">
                    <p class="toolbarFont">Sluttid</p>
                  </v-col>
                </v-row>
              </v-toolbar>
              <span v-for="(broadcast, i) in activeList" :key="i" no-action>
                <v-list class="test">
                  <v-list-item>
                    <v-list-item-content class="pa-0" id="fonts">
                      <v-row>
                        <v-col cols="3">{{broadcast.channelName ? broadcast.channelName : '?'}}</v-col>
                        <v-col cols="3">{{dateToyymmddFormat(broadcast.startDate) }}</v-col>
                        <v-col cols="3">{{dateToTimeStamp(broadcast.startDate)}}</v-col>
                        <v-col cols="3">{{dateToTimeStamp(broadcast.endDate)}}</v-col>
                      </v-row>
                    </v-list-item-content>
                  </v-list-item>
                </v-list>
              </span>
            </v-col>
          </v-container>
        </v-card-text>
        <v-card-actions>
          <v-spacer />
          <button class="btnDialogCancel" text @click="onClose()">STÄNG</button>
        </v-card-actions>
      </v-card>
    </v-dialog>
  </v-row>
</template>

<script>
import api from "../../services/BroadcastApiService";
export default {
  name: "Broadcasts",
  props: ["program"],
  data() {
    return {
      active: "preliminary",
      preliminaryBroadcasts: "",
      actualBroadcasts: "",
      dialog: false,
      metaProgramID: this.program?.metaProgramID,
      title: this.program?.originalTitle,
      season: this.program?.season,
      episode: this.program?.episode,
      broadcastList: []
    };
  },
  methods: {
    makeActive(item) {
      this.active = item;
    },
    dateToTimeStamp(item) {
      let date = new Date(item);
      return date.toLocaleTimeString("sv-SE", {
        hour: "2-digit",
        minute: "2-digit",
        hour12: false
      });
    },
    dateToyymmddFormat(item) {
      let date = new Date(item);
      return date.toLocaleDateString("sv-SE", {});
    },
    onClose() {
      this.dialog = false;
    },
    getBroadcasts() {
      api
        .get("/api/programs/" + this.metaProgramID + "/broadcasts")
        .then(response => {
          this.preliminaryBroadcasts = response.filter(
            data => data.preliminary == true
          );
          this.actualBroadcasts = response.filter(
            data => data.preliminary == false
          );
        })
        .catch(error => alert(`Error: ${error}`));
    }
  },
  computed: {
    activeList() {
      if (this.active == "preliminary") {
        return this.preliminaryBroadcasts;
      } else {
        return this.actualBroadcasts;
      }
    }
  },
  mounted() {
    this.getBroadcasts();
  }
};
</script>


<style scoped>
.preliminary,
.actual {
  padding: 8px 34px;
  border-radius: 5px;
  margin-right: 10px;
  font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
  font-weight: 100;
  transition: 0.3s;
  border: 1px solid #177cc4;
  color: #177cc4;
  margin-bottom: 15px;
}
.activeButton {
  background-color: #177cc4;
  color: cornsilk;
}

.toolbarFont {
  color: white;
}

.toolbar {
  border-radius: 5px;
}
#fonts {
  font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
  font-weight: normal;
  color: #4a4c4e;
}

.test {
  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.4);
  margin-bottom: 7px;
  border-radius: 5px;
  font-size: 13px;
  padding: 0px !important;
}
.getBroadcasts {
  font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
  font-weight: 100;
}

.btnDialogAccept,
.btnDialogCancel {
  margin-left: 10px;
  padding: 8px 34px;
  border-radius: 5px;
  margin-right: 10px;
  font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
  font-weight: 100;
  transition: 0.3s;
}
.btnDialogAccept {
  background-color: #177cc4;
  color: white;
}

.btnDialogCancel {
  border: 1px solid #177cc4;
  color: #177cc4;
}

.v-dialog {
  max-width: 700px;
}
</style>