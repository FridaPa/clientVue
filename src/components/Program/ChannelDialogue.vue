<template>
  <div>
    <v-dialog v-model="dialog" persistent max-width="100%">
      <template v-slot:activator="{on}">
        <button class="channelbutton" @click="dialog = true">Välj Kanal</button>
      </template>
      <v-card>
        <v-card-title>
          <span class="headline">Välj Kanal</span>
        </v-card-title>
        <v-card-actions>
          <v-spacer />
          <button class="btnDialogCancel" text @click="onClose()">STÄNG</button>
        </v-card-actions>
      </v-card>
      <v-card>
        <v-row>
          <v-card
            v-for="channelHouse in localChannelHouses"
            :key="channelHouse.channelHouseName"
            width="25%">
            <v-card-title>
              <v-spacer />
              <v-btn color="primary">{{channelHouse.channelHouseName}}</v-btn>
              <v-spacer />
            </v-card-title>
            <v-card-text>
              <div v-for="channel in channelHouse.channels" :key="channel.channelID">
                <v-col cols="12">
                  <v-btn
                    @click=" onSelectionChangeChannel(channel.channelID, channel.channelName)"
                  >{{channel.channelName}}</v-btn>
                </v-col>
              </div>
            </v-card-text>
          </v-card>
        </v-row>
      </v-card>
    </v-dialog>
  </div>
</template>

<script>
export default {
  name: "ChannelDialogue",
  props: {
    buttonText: String,
    localChannelHouses: {
      type: Array,
      required: true
    }
  },
  data() {
    return {
      dialog: false,
      selectedID: "",
      selectedName: ""
    };
  },
  methods: {
    onClose() {
      this.dialog = false;
    },
    onSelectionChangeChannel(id, item) {
      this.selectedName = item;
      this.selectedID = id;
      this.$emit("onSelectionChangeChannel", {
        channelName: this.selectedName,
        channelID: this.selectedID
      });
      console.log(this.selectedName, this.selectedID);
      this.onClose();
    }
  }
};
</script>

<style scoped>
.channelbutton {
  margin-left: 10px;
  padding: 8px 34px;
  border-radius: 5px;
  margin-right: 10px;
  font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
  font-weight: 100;
  border: 1px solid darkgrey;
}
</style>