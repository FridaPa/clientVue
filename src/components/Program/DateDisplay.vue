<template>
    <span>
        {{date.getFullYear()}}-{{date.getMonth() + 1}}-{{date.getDate()}}
    </span>
</template>

<script>
export default {
    props: ['date']
}
</script>