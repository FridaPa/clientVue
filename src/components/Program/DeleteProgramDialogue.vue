<template>
  <v-row justify="center">
    <v-dialog v-model="dialog" persistent max-width="700px">
      <template v-slot:activator="{on}">
        <v-icon color="red" @click="dialog = true">mdi-close-box</v-icon>
      </template>
      <v-card>
        <v-card-title>
          <span class="deleteProgram">Är du säker på att du vill ta bort detta program?</span>
        </v-card-title>
        <v-card-text>
          <v-container>
            <v-row>
              <v-col cols="11.5">
                <v-list class="test">
                  <v-list-item>
                    <v-row>
                         <v-col cols="2">{{channel}}</v-col>
                      <v-col cols="2">{{programTitle}}</v-col>
                      <v-col cols="1">{{season? season : '?'}}</v-col>
                      <v-col cols="2">{{episode? episode : '?'}}</v-col>
                      <v-col cols="2">{{genre1? genre1 :'?'}}</v-col>
                      <v-col cols="2">{{genre2? genre2 : '?'}}</v-col>
                      <v-col cols="2">{{genre3? genre3 :'?'}}</v-col>
                    </v-row>
                  </v-list-item>
                </v-list>
              </v-col>
            </v-row>
          </v-container>
        </v-card-text>
        <v-card-actions>
          <v-spacer />
          <button class="btnDialogCancel" text @click="onClose()">ÅNGRA</button>
          <button class="btnDialogAccept" text @click="selectIDforDelete(id)">JA, TA BORT</button>
        </v-card-actions>
      </v-card>
    </v-dialog>
  </v-row>
</template>

<script>
export default {
  name: "DeleteProgramDialogue",
  props: ["program"],
  components: {},
  data() {
    return {
      dialog: false,
      channel: this.program?.channelName,
      programTitle: this.program?.originalTitle,
      season: this.program?.season,
      episode: this.program?.episode,
      genre1: this.program?.genre1Name,
      genre2: this.program?.genre2Name,
      genre3: this.program?.genre3Name,
      id: this.program?.metaProgramID,
      selected: ""
    };
  },
  methods: {
    onClose() {
      this.dialog = false;
    },
    selectIDforDelete(id) {
      this.selected = id;
      this.$emit("deleteProgramById", this.selected);
      console.log(this.selected);
      this.dialog = false;
    }
  }
};
</script>

<style scoped>
.deleteProgram {
  font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
  font-weight: 100;
}
.pa-0 {
  text-align: center;
  vertical-align: middle;
}

.btnDialogAccept,
.btnDialogCancel {
  padding: 8px 34px;
  border-radius: 5px;
  margin-right: 10px;
  font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
  font-weight: 100;
  transition: 0.3s;
}
.btnDialogAccept {
  background-color: #177cc4;
  color: cornsilk;
}
.btnDialogCancel {
  border: 1px solid #177cc4;
  color: #177cc4;
}

.v-dialog {
  width: 900px !important;
}
.test {
  border: 1px solid lightgrey !important;
  margin-bottom: 7px;
  border-radius: 5px;
  font-size: 13px;
}
</style>