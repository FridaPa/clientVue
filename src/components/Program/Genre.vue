<template>
  <v-row justify="center">
    <v-dialog v-model="dialog" persistent max-width="800px">
      <template v-slot:activator="{on}">
        <button class="btnDialogue" @click="dialog = true">Välj genre</button>
      </template>
      <v-card>
        <v-card-title>
          <span class="headline">Lägg till ny Genre</span>
        </v-card-title>
        <v-card-text>
          <v-container>
            <v-row>
              <div>
                <span class="subtitle-1">Genre 1</span>
                <v-list v-for="(genre, i) in genre1" :key="i" no-action class="test">
                  <v-list-item
                    @click="selectGenre1(genre.Genre1ID, genre.Name)"
                    id="listHover" class="genre1"
                    v-bind:class="{ activeGenre: (activeGenre1 == (genre.Genre1ID)) }">
                    <v-list-item-content class="pa-0" id="fonts">
                      <v-col class="pa-0" cols="12">{{genre.Name ? genre.Name : '?'}}</v-col>
                    </v-list-item-content>
                  </v-list-item>
                </v-list>
              </div>
              <span class="space" v-if="showGenre2">
                <span class="subtitle-1">Genre 2</span>
                <v-list v-for="(genre, i) in genre2" :key="i" no-action class="test">
                  <v-list-item @click="selectGenre2(genre.Genre2ID, genre.Name)"
                    id="listHover"
                    v-bind:class="{ activeGenre: (activeGenre2 == (genre.Genre2ID)) }">
                    <v-list-item-content class="pa-0" id="fonts">
                      <v-col
                        
                        class="pa-0"
                        cols="12">{{genre.Name ? genre.Name : '?'}}
                        </v-col>
                    </v-list-item-content>
                  </v-list-item>
                </v-list>
              </span>
              <span class="space" v-if="showGenre3">
                <span class="subtitle-1">Genre 3</span>
                <v-list v-for="(genre, i) in genre3" :key="i" no-action class="test">
                  <v-list-item    @click="selectGenre3(genre.Genre3ID, genre.Name)"
                    id="listHover" 
                    v-bind:class="{ activeGenre: (activeGenre3 == (genre.Genre3ID)) }">
                    <v-list-item-content class="pa-0" id="fonts">
                      <v-col
                     
                        class="pa-0"
                        cols="12"
                      >{{genre.Name ? genre.Name : '?'}}</v-col>
                    </v-list-item-content>
                  </v-list-item>
                </v-list>
              </span>
            </v-row>
          </v-container>
        </v-card-text>
        <v-card-actions>
          <v-spacer />
          <button class="btnDialogCancel" text @click="onClose()">STÄNG</button>
        </v-card-actions>
      </v-card>
    </v-dialog>
  </v-row>
</template>

<script>
import api from "../../services/BroadcastApiService";
export default {
  name: "Genre",
  data() {
    return {
      activeGenre1: "",
      activeGenre2: "",
      activeGenre3: "",
      dialog: false,
      currentSelection: "",
      currentSelection2: "",
      showGenre2: false,
      showGenre3: false,
      genre1: [],
      genre2: [],
      genre3: [],
      genre: {
         genre1ID: "",
         genre2ID: "",
         genre3ID: "",
          name1: "",
          name2: "",
          name3: ""
          },
      
    };
  },
  methods: {
    onClose() {
      this.dialog = false;
    },
    selectGenre1(id, item) {
      this.activeGenre1 = id;
      this.genre.genre1ID = id;
      this.genre.name1 = item;
      this.genre.genre2ID = 0;
      this.genre.name2 = "";
      this.genre.genre3ID = 0;
      this.genre.name3 = "";
      api
        .get("api/genre/?id1=" + this.genre.genre1ID)
        .then(response => {
          this.genre2 = response;
        })
        .catch(error => {
          console.log(error);
        });
      this.activeGenre2 = "";
      this.showGenre2 = true;
      this.showGenre3 = false;
    },

    getGenre1() {
      api
        .get("api/genre/")
        .then(response => {
          this.genre1 = response;
        })
        .catch(error => {
          console.log(error);
        });
    },
    selectGenre2(id, item) {

      this.activeGenre2 = id;
      this.genre.genre3ID = 0;
      this.genre.name3 = "";
      this.genre.genre2ID = id;
      this.genre.name2 = item;
      api
        .get("api/genre/?id1=" + this.genre.genre1ID + "&id2=" + this.genre.genre2ID)
        .then(response => {
          this.genre3 = response;
        })
        .catch(error => {
          console.log(error);
        });
      this.showGenre3 = true;
      this.activeGenre3 = "";
       this.$emit("selectedGenre", this.genre);
            
    },
    selectGenre3(id, item) {
      console.log(this.genre);
      this.activeGenre3 = id;
      this.genre.genre3ID = id;
      this.genre.name3 = item;
      this.$emit("selectedGenre", this.genre);
      this.onClose();
    
    }
  },
  mounted: function() {
    this.getGenre1();
  }
};
</script>

<style scoped>
.space {
  margin-left: 15px;
}
#listHover:hover,
.activeGenre {
  background-color: #177cc4;
  color: white !important;
  border-radius: 5px;
}

.btnDialogAccept,
.btnDialogCancel,
.btnDialogue {
  margin-left: 10px;
  padding: 8px 34px;
  border-radius: 5px;
  margin-right: 10px;
  font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
  font-weight: 100;
  transition: 0.3s;
}
.btnDialogAccept {
  background-color: #177cc4;
  color: white;
}
.btnDialogue {
  border: 1px solid darkgray;
}

.btnDialogCancel {
  border: 1px solid #177cc4;
  color: #177cc4;
}

.v-dialog {
  width: 900px !important;
}
.test {
  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.4);
  margin-bottom: 7px;
  border-radius: 5px;
  font-size: 13px;
  width: 200px !important;
  padding: 0px !important;
}
.subtitle-1 {
  margin-bottom: 10px;
}
.v-sheet .v-sheet--title .theme--light .v-toolbar {
  border-radius: 4px;
}

.v-card {
  box-shadow: none;
  margin-left: 20px;
}
</style>