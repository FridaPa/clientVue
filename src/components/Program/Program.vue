<template>
  <v-main>
    <v-tabs slider-color="blue">
      <v-tab ripple>
        Program
      </v-tab>
      <v-tab ripple>
        Genre
      </v-tab>
      <v-tab-item>
        <v-card flat>
          <v-card-text>
               <ProgramTable/>
          </v-card-text>
        </v-card>
      </v-tab-item>
      <v-tab-item>
        <v-card flat>
          <v-card-text>
            <SeriesList/>
          </v-card-text>
        </v-card>
      </v-tab-item>
    </v-tabs>
  </v-main>
</template>

<script>


import SeriesList from "./SeriesList";
import ProgramTable from "./ProgramList.vue";
export default {
  name: "Program",
  components: {
    SeriesList,
    ProgramTable
  },  
};
</script>

<style scoped>
#align {
  vertical-align: middle;
  text-align: center;
}
.button {
  padding: 0px 40px !important;
}
.toolbarFont {
  color: white;
  font-size: 14px;
}

.toolbar {
  border-radius: 5px;
}
#fonts {
  font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
  font-weight: normal;
  color: #4a4c4e;
}

.extraContent {
  width: 98%;
  margin-left: 1%;
  background-color: red;
}

.test {
  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.4);
  margin-bottom: 7px;
  border-radius: 5px;
  font-size: 13px;
  padding: 0px !important;
}

.v-sheet .v-sheet--title .theme--light .v-toolbar {
  border-radius: 4px;
}

.v-card {
  box-shadow: none;
  margin-left: 20px;
}

.flex.offset-sm3 {
  margin-left: 5%;
}

#seriesTable div table tbody tr.v-data-table__selected {
  background-color: greenyellow;
}
</style>
