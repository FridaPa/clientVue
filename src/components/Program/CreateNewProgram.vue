<template>
  <v-row justify="center">
    <v-dialog v-model="dialog" persistent max-width="700px">
      <template v-slot:activator="{on}">
        <button class="btnDialogAccept" @click="dialog = true">Lägg till program</button>
      </template>
      <v-card>
        <v-card-title>
          <span class="createProgram">Lägg till nytt program</span>
        </v-card-title>
        <v-card-text>
          <v-container>
            <v-row>
              <v-select
                      :items="categories"
                      label="Välj kategori"
                      @change="onSelectionChangedCategory()"
                      v-model="currentSelection"
              ></v-select>
              <v-col cols="6">
                <v-text-field label="Programtitel" v-model="program.originalTitle" required/>
              </v-col>
              <v-col cols="6">
                <v-text-field label="Produktionsår" v-model="prodYear" required/>
              </v-col>
              <v-col v-if="showEpisodeandSeason" cols="6">
                <v-text-field label="Säsong" v-model="program.season" type="number" required/>
              </v-col>
              <v-col v-if="showEpisodeandSeason" cols="6">
                <v-text-field label="Episod" v-model="program.episode" type="number" required/>
              </v-col>
              <v-col cols="10">
                <v-row>
                  <Genre @selectedGenre="setSelectedGenre"/>
                  <TargetGroup @selectedCountryAndTarget="setSelectedCountryAndTarget"/>
                </v-row>
              </v-col>
            </v-row>
          </v-container>
        </v-card-text>
        <v-card-actions>
          <v-spacer/>
          <button class="btnDialogCancel" text @click="onClose()">STÄNG</button>
          <button class="btnDialogAccept" text @click="saveBroadcast()">LÄGG TILL</button>
        </v-card-actions>
      </v-card>
    </v-dialog>
  </v-row>
</template>

<script>
  import Genre from "./Genre.vue";
  import TargetGroup from "./TargetGroup.vue";

  export default {
    name: "CreateNewProgram",
    components: {Genre, TargetGroup},
    data() {
      return {
        dialog: false,
        showEpisodeandSeason: false,
        currentSelection: "",
        categories: ["Sport", "Film", "Serie"],
        prodYear: "",
        program: {
          genre1ID: "",
          genre2ID: "",
          genre3ID: "",
          genre1Name: "",
          genre2Name: "",
          genre3Name: "",
          channelName: "",
          channelID: "",
          season: "",
          episode: "",
          programCode: "",
          originalTitle: "",
          programTitle: "",
          programShortTitle: "",
          territoryCode: "",
          targetGroupCode: ""
        }
      };
    },
    methods: {
      onClose() {
        this.dialog = false;
      },
      setSelectedCountryAndTarget(value) {
        this.program.targetGroupCode = value.targetGroupCode;
        this.program.territoryCode = value.territoryCode;
      },
      setSelectedGenre(value) {
        this.program.genre1ID = value.genre1ID;
        this.program.genre2ID = value.genre2ID;
        this.program.genre3ID = value.genre3ID;
        this.program.genre1Name = value.name1;
        this.program.genre2Name = value.name2;
        this.program.genre3Name = value.name3;
      },
      saveBroadcast() {
        this.$emit("createProgram", this.program);
        this.dialog = false;
      },
      onSelectionChangedCategory() {
        if (this.currentSelection === "Serie") this.showEpisodeandSeason = true;
        else this.showEpisodeandSeason = false;
      },
    },
  };
</script>

<style scoped>
  .createProgram {
    font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
    font-weight: 100;
  }

  .btnDialogAccept,
  .btnDialogCancel {
    margin-left: 10px;
    padding: 8px 34px;
    border-radius: 5px;
    margin-right: 10px;
    font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
    font-weight: 100;
    transition: 0.3s;
  }

  .btnDialogAccept {
    background-color: #177cc4;
    color: white;
  }

  .btnDialogCancel {
    border: 1px solid #177cc4;
    color: #177cc4;
  }

  .v-dialog {
    max-width: 700px;
  }
</style>