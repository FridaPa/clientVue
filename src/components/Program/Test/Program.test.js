import Vuetify from 'vuetify'
import Vue from 'vue'
import Vuex from 'vuex'
const messageHub = new Vue();
Vue.prototype.$messageHub = messageHub;

Vue.use(Vuex);
Vue.use(Vuetify);
import { mount } from '@vue/test-utils';
import Program from "../Program.vue";
//import  {getPrograms}  from "../Program.vue";
import CreateNewProgram from '../CreateNewProgram';
//jest.mock("../Program.vue")


describe('testProgram', () => {
  window.alert = jest.fn();
  const mockUrlDelete = 'api/programs/61249';
  const mockUrl = 'api/programs/';
  const mockPrograms = [{"metaProgramID":61249,"programType":"\u0000","programCode":null,
  "channelID":204,"channelName":"Boomerang","programTitle":"Tom & Jerry","originalTitle":null,
  "programShortTitle":"Tom & Jerry","mainSerie":"Tom & Jerry","serie":null,"season":null,
  "episode":"143","countryOfOrigin":null,"productionCompany":null,"rightsOwner":null,
  "channelReference":null,"status":null,"genre1ID":4,"genre1Name":"Fiktion/Film","genre2ID":7,
  "genre2Name":"Tecknat","genre3ID":0,"genre3Name":null,"groupID":null,"targetGroupCode":"B",
  "territoryCode":"U","validationErrorID":0,"logID":0}];
   const mockProgramsDeleted = [];

 const getPrograms = jest.fn(url => mockPrograms);
  const deleteProgram = jest.fn(url => mockProgramsDeleted);

    const wrapper = mount(Program, {
      methods:{
      } 
    });

    it('renders a child', () => {
      window.alert.mockClear();
        expect(wrapper.findComponent(CreateNewProgram).exists()).toBe(true)
    })

  it('returns programs from an api call', () => {
      expect(getPrograms(mockUrl)).toBe(mockPrograms);
    });

    it('called getprograms with a mockUrl', () => {
      expect(getPrograms).toHaveBeenCalledWith(mockUrl);
    });

    it('deletes program from an api call', () => {
      expect(deleteProgram(mockUrlDelete)).toBe(mockProgramsDeleted);
    });

// mock api //
/*it('mock api real call', async () => {
  getPrograms.mockResolvedValueOnce(mockPrograms) 

   const programs = await wrapper.vm.getPrograms();
   console.log(programs);
   expect(programs).toHaveBeenCalledTimes(1);
})*/

    it('renders correctly', () => {
      window.alert.mockClear();
        expect(wrapper.element).toMatchSnapshot()
    })

    it('deletes from array', async () => {
      window.alert.mockClear();
       await wrapper.setData({programList: [
            {"metaProgramID":61242, "programTitle":"Tom & Jerry"},
            {"metaProgramID":61249, "programTitle":"Tom & Tom"},
            {"metaProgramID":61248, "programTitle":"jocke & Tom"}
        ], selectedID: 61242})
      await  wrapper.vm.deleteProgram()
      expect(wrapper.vm.$data.programList).toHaveLength(2)   
    })

    it('throws error with string insert', async () => {
      window.alert.mockClear();
       await wrapper.setData({programList: [
            {"metaProgramID":61242, "programTitle":"Tom & Jerry"},
            {"metaProgramID":61249, "programTitle":"Tom & Tom"},
            {"metaProgramID":61248, "programTitle":"jocke & Tom"}
        ], selectedID: "hej"});
   const fuc = await wrapper.vm.deleteProgram();
      expect(wrapper.vm.$data.programList).toHaveLength(3) 
  
  })
    it('adds in array', async() => {
      window.alert.mockClear();
      await  wrapper.setData({programList: [
            {"metaProgramID":61242, "programTitle":"Tom & Jerry"},
        ], program: { "metaProgramID": 46489, "programTitle":"Spacex" } })
        await  wrapper.vm.saveProgram()
        expect(wrapper.vm.$data.programList).toHaveLength(2)
    })

});
