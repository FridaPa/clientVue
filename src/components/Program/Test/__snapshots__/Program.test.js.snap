// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testProgram renders correctly 1`] = `
<div>
  <div
    class="col col-11"
  >
    <div
      class="col col-10"
    >
      <div
        class="row"
      >
        <div
          class="col-sm-6 col-md-3 col-3"
        >
          <div
            class="v-input theme--light v-text-field"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <label
                    class="v-label theme--light"
                    for="input-3"
                    style="right: 0px; position: absolute;"
                  >
                    Programtitel...
                  </label>
                  <input
                    id="input-3"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <transition-group-stub
                    class="v-messages__wrapper"
                    name="message-transition"
                    tag="div"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <br />
         
        <div
          class="col-sm-6 col-md-3 col-3"
        >
          <div
            class="v-input theme--light v-text-field"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <label
                    class="v-label theme--light"
                    for="input-6"
                    style="right: 0px; position: absolute;"
                  >
                    Säsong...
                  </label>
                  <input
                    id="input-6"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <transition-group-stub
                    class="v-messages__wrapper"
                    name="message-transition"
                    tag="div"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <br />
         
        <div
          class="col col-3"
        >
          <div
            class="v-input theme--light v-text-field"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <label
                    class="v-label theme--light"
                    for="input-9"
                    style="right: 0px; position: absolute;"
                  >
                    Episod...
                  </label>
                  <input
                    id="input-9"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <transition-group-stub
                    class="v-messages__wrapper"
                    name="message-transition"
                    tag="div"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="row justify-center"
        >
          <button
            class="btnDialogAccept"
          >
            Lägg till program
          </button>
          <div
            class="v-dialog__container"
            role="dialog"
          >
            <!---->
          </div>
        </div>
      </div>
    </div>
     
    <header
      class="toolbar v-sheet theme--light v-toolbar"
      style="height: 50px; background-color: rgb(23, 124, 196); border-color: #177cc4;"
    >
      <div
        class="v-toolbar__content"
        style="height: 50px;"
      >
        <div
          class="row"
        >
          <div
            class="align col col-1"
          >
            <p
              class="toolbarFont"
            >
              Kanal
            </p>
          </div>
           
          <div
            class="align col col-2"
          >
            <p
              class="toolbarFont"
            >
              Titel
            </p>
          </div>
           
          <div
            class="align col col-1"
          >
            <div
              class="row"
            >
              <div
                class="pa-0 col col-6"
              >
                <p
                  class="toolbarFont"
                >
                  S
                </p>
              </div>
               
              <div
                class="pa-0 col col-6"
              >
                <p
                  class="toolbarFont"
                >
                  E
                </p>
              </div>
            </div>
          </div>
           
          <div
            class="col col-4"
          >
            <div
              class="row"
            >
              <div
                class="pa-0 col col-4"
              >
                <p
                  class="toolbarFont"
                >
                  Genre 1
                </p>
              </div>
               
              <div
                class="pa-0 col col-4"
              >
                <p
                  class="toolbarFont"
                >
                  Genre 2
                </p>
              </div>
               
              <div
                class="pa-0 col col-4"
              >
                <p
                  class="toolbarFont"
                >
                  Genre 3
                </p>
              </div>
            </div>
          </div>
           
          <div
            class="align col col-1"
          >
            <div
              class="row"
            >
              <div
                class="pa-0 col col-6"
              >
                <p
                  class="toolbarFont"
                >
                  M
                </p>
              </div>
               
              <div
                class="pa-0 col col-6"
              >
                <p
                  class="toolbarFont"
                >
                  L
                </p>
              </div>
            </div>
          </div>
           
          <div
            class="align col col-3"
          />
        </div>
      </div>
    </header>
     
  </div>
</div>
`;
